portal/templates/profile_macros.html

```diff
@@ -2232,3 +2235,342 @@ <h4 id="clinicalDataLoc" class="profile-item-title index-item">{% if person.id =
     <button id="updateProfile" type="submit" class="btn btn-lg btn-tnth-primary">{{ _("Save") }}</button>
     <span id="confirmMsg" class="text-muted tnth-hide small-text">&nbsp;&nbsp;<em>{{ _("Profile changes have been saved") }}</em></span>
 {%- endmacro %}
+{% macro accountCreationScript() -%}
+    <script>
+    var AccountCreationObj = function (roles) {
+        this.attempts = 0;
+        this.max_attempts = 3;
+        this.params = null;
+        this.roles = roles ? roles : [{'name': '{{ROLE.PATIENT}}'}, {'name': '{{ROLE.WRITE_ONLY}}'}];
```

Comment:
Not sure I love defaulting to PATIENT & WRITE_ONLY.  Shouldn't we just enforce passing the correct set of roles based on the context?